### PR TITLE
feat: support cjs and esm

### PR DESCRIPTION
**Pull request by italoteix**
Aiming for multiple targets will allow us to use this addon for tests with `@storybook/testing-react` as it runs with `jest`.

### CODE REVIEW

**Comment by bhishp** [comment]
Hey, thanks for this, I'll get a chance to check it out today. I see you're still making commits, let me know when you're happy with it/finalised

**Comment by italoteix** [comment]
> Hey, thanks for this, I'll get a chance to check it out today. I see you're still making commits, let me know when you're happy with it/finalised

Thanks for your time. I'm still doing some tests on my app to check if it's 100% working. But at least on the provided examples, it is working.

**Comment by italoteix** [comment]
@bhishp It seems to be working, but I'm thinking on add some tests to show the use case. Do you mind if I do the `jest` and `testing-library` setup?

**Comment by italoteix** [comment]
@bhishp  I think that now I'm done with this PR. Can you check it now?

File: stories/fields.test.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+// And it also works for testing your components with @storybook/testing-react
+describe('subforms', () => {
+  test('works works', () => {
+    render(<MyTextInput />);
+    expect(
+      screen.getByText('Describe formik in 80 characters')
+    ).toBeInTheDocument();
+  });
+});
```
Comment (yannbf) [inline stories/fields.test.tsx:17]:
```suggestion
describe('subforms', () => {
  // Stories are reused and the tests become simpler thanks to https://github.com/storybookjs/testing-react
  test('renders correctly from the story using Formik decorator', () => {
    render(<MyTextInput />);
    expect(
      screen.getByText('Describe formik in 80 characters')
    ).toBeInTheDocument();
  });
});
```

**Review by bhishp — APPROVED** [review]
This is very cool, thanks for the addition

**Comment by bhishp** [comment]
I have just released this feature (along with several dependency upgrades) in `v2.4.0`. Let me know if you encounter any issues. Thanks again for your contribution

https://www.npmjs.com/package/storybook-formik/v/2.4.0

**Pull request merged**